File: python/easy_challenges/fibonacci_series/fibonacciSeries.py
'''
Program: fibonacciSeries.py
Author: Seth Tucker
Date: 10/21/2016
Last Modified: 10/21/2016

'''

#ERROR: When submited to CodeEval score is only 95
#There's an error somwhere

import sys

def computeFib(number):
  if number == 0:
    fibNum = 0
  elif number == 1:
    fibNum = 1
  else:
    fibNum = 0
    digitOne = 1
    digitTwo = 1
    for i in range(2, number):
      fibNum = digitOne + digitTwo
      digitOne = digitTwo
      digitTwo = fibNum
  return fibNum

def main():
  file = open(sys.argv[1], 'r')
  for number in file:
    number = number.strip()
    if len(number) == 0:
      continue
    print(computeFib(int(number)))

  file.close()

main()

  

File: c++/easy_challenges/fibonacci_series/fibonacciSeries.cpp
/*
Program: fibonacciSeries.cpp
Author: Seth Tucker
Date: 9/5/2016
Last Modified: 11/9/2016

Program takes input from a text file and computes the nth Fibonacci 
number where n is pulled from the text file. 
Ex: n = 5, nth Fibonacci number = 5
    n = 12, nth Fibonacci number = 144
*/

#include <iostream>
#include <string>
#include <fstream>
#include <vector>

using std::cout;
using std::endl;
using std::ifstream;
using std::string;
using std::vector;

//Compute nth Fibonacci number
long computeFib(long number){
	long fibNum = 0;
	//If number = 0
	if (number == 0) fibNum = 0;
	//If number = 1 or number = 2
	else if (number == 1 || number == 2) fibNum = 1;
	//Everything > 2
	else {
		long digitOne = 1;
		long digitTwo = 1;
		for (auto j = 2; j < number; j++) {
			fibNum = digitOne + digitTwo;
			digitOne = digitTwo;
			digitTwo = fibNum;
		}
	}
	return fibNum;
	
}  //End computeFib()

#ifndef UNIT_TESTING

int main(int argc, char *argv[]) {
	string fileName(argv[1]);
	ifstream source;
	source.open(fileName);
	
	//Get numbers from text file and store in a vector
	vector<long> numbers;
	if (source) {
		long value;
		while (source >> value) {
			numbers.push_back(value);
		}
	}
	source.close();

	//Compute nth Fibonacci number
	for (auto i = 0; i < numbers.size(); i++) {
		long fibNum = computeFib(numbers[i]);
		cout << fibNum << endl;
	}

	return 0;

}  //End main()


#endif